### PR TITLE
[primTorch] Adds dsplit/dstack reference

### PR DESCRIPTION
**Pull request by jjsjann123**
Added references _refs.dsplit & _refs.dstack and PythonRefInfo tests

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -2055,6 +2063,14 @@ def tensor_split(
         return tuple(splits)
 
 
+def dsplit(a: TensorLikeType, sections: DimsType) -> TensorSequenceType:
+    if a.ndim < 3:
+        raise RuntimeError(f"torch.dsplit requires a tensor with at least 3 dimension, but got a tensor with {a.ndim} dimensions!")
+    if isinstance(sections, int) and (sections == 0 or a.shape[2] % sections != 0):
```
Comment (mruberry) [inline torch/_refs/__init__.py:2069]: Should this add a check that sections is strictly positive?

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -2055,6 +2063,14 @@ def tensor_split(
...
+        raise RuntimeError(f"torch._refs.dsplit attempted to split along dimension 2, but the size of the dimension {a.shape[2]} is not divisible by the split_size {sections}!")
```
Comment (mruberry) [inline torch/_refs/__init__.py:2070]: Is this error check redundant with calling tensor_split?

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -19762,6 +19762,14 @@ def __init__(
         "_refs.column_stack",
         torch_opinfo_name="column_stack",
     ),
+    PythonRefInfo(
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:19765]: Nice

**Comment by mruberry** [comment]
Lint's going to need a fix but otherwise I think this looks great, @jjsjann123 -- curious about your thoughts on error checking redundancy, however

File: torch/_refs/__init__.py (hunk 2 above)
Comment (jjsjann123) [reply torch/_refs/__init__.py:2070]: it is actually not.

`a.shape[2] % sections != 0` this should only apply to dsplit. (double checked with numpy as well as torch)
`sections == 0` <- this one will actually also error out on tensor_split, but error message is not consistent with `torch.dsplit`

File: torch/_refs/__init__.py (hunk 1 above)
Comment (mruberry) [reply torch/_refs/__init__.py:2071]: No -- we should probably introduce an error context, instead, that can produce nicer error messages without developers doing all that work.

Comment (jjsjann123) [reply torch/_refs/__init__.py:2071]: IIUC, #78738 tracks this error message problem in the long term.

For the scope in this PR, I think I'm striking a sweet balance here: some dsplit specific checks are done here, while relying some checks (non negative sections) in the callee.

Do we still have any blocking concern on merging this PR?

Comment (mruberry) [reply torch/_refs/__init__.py:2085]: No -- sounds great

**Review by mruberry — APPROVED** [review]
Cool! Good to land as soon as the test jobs pass. Thanks @jjsjann123!

**Pull request closed without merging**